modular-docs-manual/content/topics/concept_using-a-document-attribute-in-an-anchor-id.adoc

```diff
@@ -0,0 +1,86 @@
+[#using-a-document-attribute-in-an-anchor-id]
+
+= Using a Document Attribute in an Anchor ID
+
+In some cases, a book will need to include the same module multiple times within the same assembly.
+How do you reuse modules within the same assembly without getting build errors for using duplicate anchor IDs?
+The answer is embedding a user-defined document attribute in the anchor ID name.
+
+Programming languages call this variable substitution, which is also know as variable interpolation.
+The AsciiDoc markup language is doing the same type of substitution with document attributes.
+
+There are three syntax formats for creating anchor IDs within a module or an assembly:
+
+* *Standard ID Syntax:*
++
+----
+[[anchor-id]]
+----
+
+* *Short-hand ID Syntax:*
++
+----
+[#anchor-id]
+----
+
+* *Long-hand ID Syntax:*
++
+----
+[id='anchor-id']
+----
+
+[discrete]
+== Examples
+
+Only the short-hand and long-hand ID syntax formats support document attribute substitution.
+
+.Syntax Example
+----
+[#anchor-id-{doc_attribute}]
+----
+or
+----
+[id='anchor-id-{doc_attribute}']
+----
+
+Define the document attribute in the assembly, such as the `master.adoc` file.
+
+////
+TODO - ritz303 : I've only seen this appendix rendering problem in one scenario.
+Need to do some more research and testing in different scenarios to verify the validity of the note below before adding this content.
+
+[NOTE]
+====
+When including an appendix in a book, and using either the short-hand or the long-hand ID syntax format, will cause the appendix to be rendered as a chapter and not an appendix.
+Using the standard syntax format will render the appendix correctly.
+====
+////
+
+.Practical Example
+
+In a book about data replication, you want to reuse a procedure module for monitoring the progress of data replication in two different chapters.
+
+The assembly file contains the following content:
+----
+= Data Replication Guide
+
+// Chapter 1 - Doing the Initial Data Replication of Disks
+:chapter: initial
+\include::procedure_monitoring-the-progress-of-data-replication.adoc[leveloffset=+1]
+
+// Chapter 2 - Replacing a Fail Disk
+:chapter: replacing
+\include::procedure_monitoring-the-progress-of-data-replication.adoc[leveloffset=+1]
+----
+
+The first two lines of the _procedure_monitoring-the-progress-of-data-replication.adoc_ module file contains the following content:
+----
+[id='monitoring-the-progress-of-data-replication-{chapter}']
+
+= Monitoring the Progress of Data Replication
+----
+
+[discrete]
+== Additional Resources
+
+* The link:http://asciidoctor.org/docs/user-manual/[AsciiDoc User Manual].
```

Comment:
s/AsciiDoc/Asciidoctor

Reply:
@phayz : Good catch!  I've updated the source content.

Reply:
Now that I understand how this works, the instructions as written make sense to me. But when I read them the first time, I struggled to understand how to implement this and it took me many tries to get it right. I would rewrite this content lto be something like the following. I'd be happy to update the asciidoc file if you want me to. 
### 
Reusing a Module in an Assembly

To use a module multiple times in an assembly, you must employ a document attribute that distinguishes each instance of the reused module. Without the distinguishing document attribute, there are build errors because the same anchor ID appears multiple times in a book.

Follow these steps:

1. In the module that is being reused, define the anchor ID in one of these formats:

[#add-eggs]
[id='add-eggs']

You cannot use this format: [[add-eggs]]

2. Append a document attribute to the anchor ID. For example, suppose the name of the document attribute is **reuse**. In the module being reused, the anchor ID would be one of these:

[#add-eggs-{reuse}]
[id='add-eggs-{reuse}]

3. In an assembly that includes the module being reused, specify a value for, for example, the reuse document attribute. Do this after the heading and before the first paragraph. For example:

[[bake-cake]]
= Bake a Cake

:reuse: cake
The introductory paragraph goes here and it might have a cross-reference to the module being reused and that reference would look something like this: see <<add-eggs-{reuse}>>. Or maybe this paragraph introduces a bulleted list of cross-references. Any one of the cross-references can look like this:

. <<add-eggs-{reuse}>>.

And then of course since this is an assembly, it would have an include statement for the module being reused. There's nothing special about the include statement. For example:

include::add-eggs.adoc[leveloffset=+1]

4. Repeat the previous step for each assembly that includes the module being reused. In each assembly, specify a different value for the reuse document attribute. For example:

[[bake-muffins]]
= Bake Muffins

:reuse: muffin
Your text starts here and it can contain a cross-reference that would look like this:

See <<add-eggs-{reuse}>>

include::add-eggs.adoc[leveloffset=+1]